x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_package_custom_extension/endpoint_package_custom_extension.tsx

```diff
@@ -27,91 +30,147 @@ import {
+    () => EventFiltersApiClient.getInstance(http),
+    [http]
+  );
+
+  return (
+    <FleetArtifactsCard
+      {...props}
+      artifactApiClientInstance={eventFiltersApiClientInstance}
+      getArtifactsPath={getEventFiltersListPath}
+      labels={EVENT_FILTERS_LABELS}
+      data-test-subj="eventFilters"
+    />
+  );
+});
+EventFiltersArtifactCard.displayName = 'EventFiltersArtifactCard';
+
+const HostIsolationExceptionsArtifactCard = memo<PackageCustomExtensionComponentProps>((props) => {
+  const http = useHttp();
+  const hostIsolationExceptionsApiClientInstance = useMemo(
+    () => HostIsolationExceptionsApiClient.getInstance(http),
+    [http]
+  );
+
+  return (
+    <FleetArtifactsCard
+      {...props}
+      artifactApiClientInstance={hostIsolationExceptionsApiClientInstance}
+      getArtifactsPath={getHostIsolationExceptionsListPath}
+      labels={HOST_ISOLATION_EXCEPTIONS_LABELS}
+      data-test-subj="hostIsolationExceptions"
+    />
+  );
+});
+HostIsolationExceptionsArtifactCard.displayName = 'HostIsolationExceptionsArtifactCard';
+
+const BlockListArtifactCard = memo<PackageCustomExtensionComponentProps>((props) => {
+  const http = useHttp();
+  const blocklistsApiClientInstance = useMemo(() => BlocklistsApiClient.getInstance(http), [http]);
+
+  return (
+    <FleetArtifactsCard
+      {...props}
+      artifactApiClientInstance={blocklistsApiClientInstance}
+      getArtifactsPath={getBlocklistsListPath}
+      labels={BLOCKLISTS_LABELS}
+      data-test-subj="blocklists"
+    />
+  );
+});
+BlockListArtifactCard.displayName = 'BlockListArtifactCard';
+
+/**
+ * The UI displayed in Fleet's Endpoint integration page, under the `Advanced` tab
+ */
 export const EndpointPackageCustomExtension = memo<PackageCustomExtensionComponentProps>(
   (props) => {
-    const http = useHttp();
-    const { loading, canAccessEndpointManagement, canReadHostIsolationExceptions } =
-      useEndpointPrivileges();
-
-    const trustedAppsApiClientInstance = useMemo(
-      () => TrustedAppsApiClient.getInstance(http),
-      [http]
-    );
-
-    const eventFiltersApiClientInstance = useMemo(
-      () => EventFiltersApiClient.getInstance(http),
-      [http]
-    );
-
-    const hostIsolationExceptionsApiClientInstance = useMemo(
-      () => HostIsolationExceptionsApiClient.getInstance(http),
-      [http]
-    );
-
-    const blocklistsApiClientInstance = useMemo(
-      () => BlocklistsApiClient.getInstance(http),
-      [http]
-    );
-
-    const artifactCards = useMemo(
-      () => (
+    const {
+      loading,
+      canReadBlocklist,
+      canReadEventFilters,
+      canReadTrustedApplications,
+      canReadHostIsolationExceptions,
+    } = useEndpointPrivileges();
+    const { docLinks } = useKibana().services;
+
+    const userCanAccessContent = useCanAccessSomeArtifacts();
+
+    const artifactCards: ReactElement = useMemo(() => {
+      if (loading) {
+        return <></>;
```

Comment:
Should we display a spinner here instead?

Reply:
Good call. Will add it.